ecs-cli/modules/cli/configure/configure_test.go

```diff
@@ -15,125 +15,99 @@ package configure
+func createProfileConfig(name string) *cli.Context {
+	flags := flag.NewFlagSet("ecs-cli", 0)
+	flags.String(command.AccessKeyFlag, name, "")
+	flags.String(command.SecretKeyFlag, awsSecretKey, "")
+	flags.String(command.ProfileNameFlag, name, "")
+	return cli.NewContext(nil, flags, nil)
 }
 
-func TestConfigInitWithPrefixes(t *testing.T) {
-	setPrefixes := flag.NewFlagSet("ecs-cli", 0)
-	setPrefixes.String(command.ProfileFlag, profileName, "")
-	setPrefixes.String(command.ClusterFlag, clusterName, "")
-
-	composeProjectName := "projectName"
-	composeServiceName := "serviceName"
-	cfnStackName := "stackName"
-
-	setPrefixes.String(command.ComposeProjectNamePrefixFlag, composeProjectName, "")
-	setPrefixes.String(command.ComposeServiceNamePrefixFlag, composeServiceName, "")
-	setPrefixes.String(command.CFNStackNamePrefixFlag, cfnStackName, "")
-
-	context := cli.NewContext(nil, setPrefixes, nil)
+func TestClusterConfigureAndSetDefault(t *testing.T) {
+	// Config init when just cluster and region are specified
+	config1 := createClusterConfig(profileName)
+	config2 := createClusterConfig(secondProfile)
+	// Create a temprorary directory for the dummy ecs config
+	tempDirName, err := ioutil.TempDir("", "test")
+	if err != nil {
+		t.Fatal("Error while creating the dummy ecs config directory")
+	}
+	os.Setenv("HOME", tempDirName)
+	defer os.Unsetenv("HOME")
+	defer os.RemoveAll(tempDirName)
+
+	// configure 2 profiles and set one as default
+	ConfigureCluster(config1)
+	ConfigureCluster(config2)
+	ConfigureDefaultCluster(config2)
+
+	parser, err := config.NewReadWriter()
+	assert.NoError(t, err, "Error reading config")
+	readConfig, err := parser.Get("", "")
+	assert.NoError(t, err, "Error reading config")
+	assert.Equal(t, region, readConfig.Region, "Region mismatch in config.")
+	assert.Equal(t, secondProfile, readConfig.Cluster, "Cluster name mismatch in config.")
+	assert.Empty(t, readConfig.ComposeProjectNamePrefix, "Compose project prefix name should be empty.")
+	assert.Empty(t, readConfig.ComposeServiceNamePrefix, "Compose service prefix name should be empty.")
+	assert.Empty(t, readConfig.CFNStackNamePrefix, "CFNStackNamePrefix should be empty.")
 
-	cfg, err := createECSConfigFromCLI(context)
-	assert.NoError(t, err, "Unexpected error reading config from rdwr")
-	assert.Equal(t, composeProjectName, cfg.ComposeProjectNamePrefix, "Expected ComposeProjectName to match in config")
-	assert.Equal(t, composeServiceName, cfg.ComposeServiceNamePrefix, "Expected ComposeServiceName to match in config")
-	assert.Equal(t, cfnStackName, cfg.CFNStackNamePrefix, "Expected CfnStackName to match in config")
 }
 
-func TestConfigInitWithoutPrefixes(t *testing.T) {
-	setNoPrefixes := flag.NewFlagSet("ecs-cli", 0)
-	setNoPrefixes.String(command.ProfileFlag, profileName, "")
-	setNoPrefixes.String(command.ClusterFlag, clusterName, "")
-
-	context := cli.NewContext(nil, setNoPrefixes, nil)
+func TestProfileConfigureAndSetDefault(t *testing.T) {
```

Comment:
Can you add the following error cases?
* Set default profile that does not exist
* Set cluster profile without a cluster
* Set cluster profile without a region
* Set cred profile without secret
* Set cred profile without access key